### DISCUSSION TITLE
force XWayland to start when compositor starts?

### DISCUSSION THREAD

**Discussion by RicArch97** [labwc/labwc#1978] [Q&A] [answered]
I noticed the new option `xwaylandPersistence` was added, but it seems this is about persisting an already existing XWayland instance.

What i'd be interested in is to force start XWayland when LabWC starts, similar to Sway's `xwayland force` option.

My use case is that with multiple monitors, I have to set the "primary display" for XWayland using e.g. `xrandr --output DP-1 --primary`, so that games select the correct monitor. This is best done when the compositor starts (e.g. a startup script), but the `xrandr` command has no effect when XWayland is not started yet.

What do you think?

**Comment by tokyo4j** [comment]
@ahesford Does #1963 address this?

**Comment by ahesford** [comment, marked as answer]
The XWayland persistence change should be sufficient to address this issue. In my testing, enabling persistence causes wlroots to launch the X server right away and, in any case, invoking xrandr in the autostart should be sufficient to trigger the launch of a persistent server if the option is enabled.

Support for xinitrc may or may not allow an automatic xrandr command to perform the desired configuration, depending on the timing of the client's actions.

**Reply by tokyo4j** [reply]
It sure looks like it. I confirmed sway's `xwayland force` behaves in the same way as `<allowPersistence>yes</allowPersistence>`, both in my testing and from the source code.

**Comment by Consolatis** [comment]
This might also be related to the outputs not being available by the time xrandr is called.
